### PR TITLE
[Merged by Bors] - add render_to_texture example

### PR DESCRIPTION
**Pull request by mrk-its**

### CODE REVIEW

File: examples/3d/render_to_texture.rs [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+        .insert(FirstPass)
+        .remove::<MainPass>();
+    // light
+    commands.spawn_bundle(PointLightBundle {
+        transform: Transform::from_translation(Vec3::new(0.0, 0.0, 10.0)),
+        ..Default::default()
+    });
+    // camera
+
+    let mut first_pass_camera = PerspectiveCameraBundle {
```
Comment (MinerSebas) [inline examples/3d/render_to_texture.rs:234]:
```suggestion

    // camera
    let mut first_pass_camera = PerspectiveCameraBundle {
```

**Review by MinerSebas — COMMENTED** [review]
Please also add the example to the README.md in the examples folder.

**Comment by alice-i-cecile** [comment]
This is half-way to #22, correct?

**Comment by mrk-its** [comment]
> This is half-way to #22, correct?

Yes. The TextureNode from this PR may be used as replacement of WindowSwapChainNode, I've created POC: https://github.com/mrk-its/bevy/blob/render_to_file/examples/3d/render_to_file.rs.

**Comment by mrk-its** [comment]
> Please also add the example to the README.md in the examples folder.

done

**Comment by mtsr** [comment]
Thank you for this, @mrk-its!

I'm already thinking how we can add a post-process pass, once this is merged. I already have an upscale + post-process pass in my own prototype, but it's probably a bit of work to make it generic enough to PR.

**Comment by cart** [comment]
Looks good to me! Just made some small tweaks to the trait / components in the example to cut down on boilerplate and be a bit more explicit. (As an aside, I'd like to consider moving away from the "graph extension trait" pattern internally as well. I think functions work fine, but we could also consider a single "graph plugin" trait as an alternative)

**Pull request closed without merging**